src/state/ResourceCreationState.ts

```diff
@@ -0,0 +1,256 @@
+
+/**
+ * Represents the state of resource creation.
+ * This class is responsible for initializing the ResourceCreationState object.
+ *
+ * Uses {@link accounts} and {@link tokens} from the initialResources.json
+ * to create initial resources for the local-node environment
+ * 
+ * @implements {IState}
+ */
+export class ResourceCreationState implements IState {
+    /**
+     * The name of the state.
+     */
+    private readonly stateName: string;
+
+    /**
+     * The logger used for logging resource creation state information.
+     */
+    private readonly logger: LoggerService;
+
+    /**
+     * The CLI service used for resource creation.
+     */
+    private readonly cliService: CLIService;
+
+    /**
+     * The client service used for resource creation.
+     */
+    private readonly clientService: ClientService;
+
+    /**
+     * The observer for the resource creation state.
+     */
+    private observer: IOBserver | undefined;
+
+    /**
+     * Represents the state of resource creation.
+     * This class is responsible for initializing the ResourceCreationState object.
+     */
+    constructor() {
+        this.stateName = ResourceCreationState.name;
+        this.logger = ServiceLocator.Current.get<LoggerService>(LoggerService.name);
+        this.cliService = ServiceLocator.Current.get<CLIService>(CLIService.name);
+        this.clientService = ServiceLocator.Current.get<ClientService>(ClientService.name);
+        this.logger.trace('Resource Creation State Initialized!', this.stateName);
+    }
+
+    /**
+     * Subscribes an observer to receive updates from the ResourceCreationState.
+     * @param {IOBserver} observer The observer to subscribe.
+     */
+    public subscribe(observer: IOBserver): void {
+        this.observer = observer;
+    }
+
+    /**
+     * This method is responsible for starting the ResourceCreationState.
+     * It creates tokens asynchronously or synchronously based on the CLI arguments.
+     * @returns Promise that resolves when the state is started.
+     * @emits {EventType.Finish} When the state is finished.
+     */
+    public async onStart(): Promise<void> {
+        const { async } = this.cliService.getCurrentArgv();
+        
+        const mode = async ? 'asynchronous' : 'synchronous';
+        this.logger.info(
+          `Starting Resource Creation State in ${mode} mode`, this.stateName);
+
+        const promise = this.createResources()
+          .then(() => this.observer!.update(EventType.Finish));
+        if (!async) {
+            await promise;
+        }
+    }
+
+    /**
+     * Creates accounts and tokens with the given properties and associates them.
+     * @returns Promise that resolves when all resources are created.
+     */
+    private async createResources(): Promise<void> {
+        const accountProps: IAccountProps[] = accounts as unknown as IAccountProps[];
+        const tokenProps: ITokenProps[] = tokens as unknown as ITokenProps[];
```

Comment:
why go though unknown and then to specific type, why not directly as correct type ?